File: Assignments/Assignment 38/C Codes/program38_1.c
// 1.Write a program which accept string from user and copy that characters of that string 
// in another string in reverse order.

// Input : "Marvellous Multi OS"

// Output: OS itluM suollevraM

#include<stdio.h>

void StrCpyRev(char *str, char *dest)
{
    char *temp1 = NULL;
    char *temp2 = NULL;

    temp1 = str;
    temp2 = dest;

    while(*str != '\0')
    {
        str++;
    }

    str--;

    while(temp1 < str)
    {
        *dest = *str;
        str--;
        dest++;
    }

    *dest = *temp1;
    dest++;
    *dest = '\0';

    printf("%s\n", temp2);
}

int main()
{
    char arr[30] = "Marvellous Python";
    
    char brr[30];

    StrCpyRev(arr, brr);

    return 0;
}

File: Assignments/Assignment 38/Java Codes/program38_1.java
// 1.Write a program which accept string from user and copy that characters of that string 
// in another string in reverse order.

// Input : "Marvellous Multi OS"

// Output: OS itluM suollevraM

class Test
{
    String StrCpyRev(String str)
    {
        char Arr[] = str.toCharArray();
        int iCnt = 0, iBrr = 0;
        char temp = '\0';

        char Brr[] = new char[Arr.length];

        for(iCnt = (Arr.length-1), iBrr = 0; iCnt >= 0; iCnt--, iBrr++)
        {
            Brr[iBrr] = Arr[iCnt];
        }

        return new String(Brr);
    }
}

class program38_1
{
    public static void main(String A[])
    {
        String Arr = "Marvellous Python";

        String Brr = null;

        Test tobj = new Test();

        Brr = tobj.StrCpyRev(Arr);

        System.out.println(Brr);
    }
}